### DISCUSSION TITLE
ArangoDB?

### DISCUSSION THREAD

**Discussion by jackgray** [Budibase/budibase#5756] [General]
Hi everyone. I guess I didn't do deep enough investigation before designing my solution architecture, because I based it on the fact that Budibase claims to support ArangoDB, but now that I am at the actual integration stage, I can't seem to find a single instance of anyone on the web using them together, much less any documentation for it...

I have both services successfully running independently on docker, and can access their endpoints and connect the Arango one as a Budibase backend without any error. However I am not sure how to create a query or access the data through Budibase in any way. 

I try using the straightforward AQL query of " FOR item in <collection_name> RETURN item" and receive a timeout error. When working against the native Arango web interface or python API, the query takes only a second, so it seems unlikely that it simply needs more time. I trust the budibase developers to include a reasonable timeout setting in the default compose settings. I will try increasing, but am banking on that being a dead end. 

What is the support status of the ArangoDB integration, and does anyone know of it being used successfully?

Thanks!

**Comment by melohagan** [comment]
Hi @jackgray 

This integration was added by a collaborator over a year ago: https://github.com/Budibase/budibase/pull/1036

I don't think the core dev team has worked with this integration much. 

I did give it arango, but I was getting the same issue. Seems to be trying to connect on localhost regardless of what is typed into the url field:
`Error querying arangodb connect ECONNREFUSED 127.0.0.1:8529`

I've created an issue from this discussion here: https://github.com/Budibase/budibase/issues/5782

**Reply by kashifraja** [reply]
IHi,

After successfully got connected budibase with arangodb. Now i am able to run Query below.

FOR doc INTO Client RETURN doc 

all records showing but to insert from AQL does not work, anyone can assist pls
